homeassistant/components/wolflink/sensor.py

```diff
@@ -29,37 +41,90 @@ async def async_setup_entry(
     async_add_entities: AddConfigEntryEntitiesCallback,
 ) -> None:
     """Set up all entries for Wolf Platform."""
-
     coordinator = hass.data[DOMAIN][config_entry.entry_id][COORDINATOR]
     parameters = hass.data[DOMAIN][config_entry.entry_id][PARAMETERS]
     device_id = hass.data[DOMAIN][config_entry.entry_id][DEVICE_ID]
 
-    entities: list[WolfLinkSensor] = []
-    for parameter in parameters:
-        if isinstance(parameter, Temperature):
-            entities.append(WolfLinkTemperature(coordinator, parameter, device_id))
-        if isinstance(parameter, Pressure):
-            entities.append(WolfLinkPressure(coordinator, parameter, device_id))
-        if isinstance(parameter, PercentageParameter):
-            entities.append(WolfLinkPercentage(coordinator, parameter, device_id))
-        if isinstance(parameter, ListItemParameter):
-            entities.append(WolfLinkState(coordinator, parameter, device_id))
-        if isinstance(parameter, HoursParameter):
-            entities.append(WolfLinkHours(coordinator, parameter, device_id))
-        if isinstance(parameter, SimpleParameter):
-            entities.append(WolfLinkSensor(coordinator, parameter, device_id))
+    entities: list[WolfLinkSensor] = [
+        WolfLinkSensor(
+            coordinator, parameter, device_id, get_entity_description(parameter)
+        )
+        for parameter in parameters
+    ]
 
     async_add_entities(entities, True)
 
 
+def get_entity_description(parameter: Parameter) -> SensorEntityDescription:
+    """Return the entity description for a given parameter."""
+    if isinstance(parameter, Temperature):
+        return SensorEntityDescription(
+            key=parameter.parameter_id,
+            name=parameter.name,
+            device_class=SensorDeviceClass.TEMPERATURE,
+            native_unit_of_measurement=UnitOfTemperature.CELSIUS,
```

Comment:
```suggestion
            native_unit_of_measurement=UnitOfTemperature.CELSIUS,
            support_fn= lambda parameter: isinstance(parameter, Temperature)
```
make this just a list of `WolflinkSensorEntityDescriptions`

where 
```python
@dataclass(kw_only=True, frozen=True)
class WolflinkSensorEntityDescriptions(SensorEntityDescription):
    supported_fn: Callable[[Whatever base class parameter has], bool]
```

then you can check during the looping of the entities `if entity_description.supported_fn(parameter)`